compiler/rustc_resolve/src/lib.rs

```diff
@@ -2529,7 +2530,20 @@ impl<'a> Resolver<'a> {
 
                             (format!("use of undeclared type `{}`", ident), suggestion)
                         } else {
-                            (format!("use of undeclared crate or module `{}`", ident), None)
+                            (
+                                format!("use of undeclared crate or module `{}`", ident),
+                                self.find_similarly_named_module_or_crate(
+                                    ident.name,
+                                    &parent_scope.module,
+                                )
+                                .map(|sugg| {
+                                    (
+                                        vec![(ident.span, sugg.to_string())],
+                                        String::from("a similar crate or module exists"),
```

Comment:
```suggestion
                                        String::from("there is a crate or module with a similar name"),
```

Ideally we would get the type of what the `ident` is from `find_similarly_named_module_or_crate` so we can be more specific in the message. We can do that later.

Reply:
Ok. I'll do it in a new PR.